internal/providers/session/store.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package session
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stacklok/minder/internal/db"
+)
+
+type SessionStore interface {
+	GetBySessionState(ctx context.Context, state string) (*db.GetProjectIDBySessionStateRow, error)
+}
+
+type sessionStore struct {
+	store db.Store
+}
+
+func NewSessionStore(store db.Store) SessionStore {
+	return &sessionStore{store: store}
+}
+
+func (s *sessionStore) GetBySessionState(ctx context.Context, state string) (*db.GetProjectIDBySessionStateRow, error) {
```

Comment:
Unless you plan on adding more methods, I don't think this interface/struct serves much purpose. If you need to stub a single method of the store, you could keep the interface, but use `db.Store` as the implementation.

Reply:
I agree. I was planning on initially but then it turned I don't need this interface.